bciers/apps/registration/app/components/facilities/FacilitiesForm.tsx

```diff
@@ -0,0 +1,76 @@
+"use client";
+
+import { useState } from "react";
+import { useParams, useRouter } from "next/navigation";
+import SingleStepTaskListForm from "@bciers/components/form/SingleStepTaskListForm";
+import { actionHandler } from "@bciers/utils/actions";
+
+export interface FacilityFormData {
+  [key: string]: any;
+}
+
+interface Props {
+  schema: any;
+  uiSchema: any;
+  formData?: FacilityFormData;
+  disabled?: boolean;
+}
+
+export default function FacilitiesForm({
+  formData,
+  schema,
+  uiSchema,
+  disabled,
+}: Readonly<Props>) {
+  // @ ts-ignore
+  const [error, setError] = useState(undefined);
+  const [confirmation, setConfirmation] = useState(false);
+  const router = useRouter();
+  const params = useParams();
+  let isCreating = formData && Object.keys(formData).length > 0 ? false : true;
+  return (
+    <>
+      {confirmation && <div>success</div>}
+      <SingleStepTaskListForm
+        disabled={disabled}
+        schema={schema}
+        uiSchema={uiSchema}
+        formData={formData}
+        onSubmit={async (data: { formData?: any }) => {
+          const method = isCreating ? "POST" : "PUT";
+          const endpoint = isCreating ? "registration/facilities" : `tbd`;
+          const pathToRevalidate = isCreating ? "dashboard/facilities" : `tbd`;
+          // brianna bug potential this is going to overwrite with nesting, going to have to do some processing here
+          const body = {
+            ...formData?.section1,
+            ...formData?.section2,
+            ...data.formData.section1,
+            ...data.formData.section2,
+            operation_id: params.operationId,
+          };
```

Comment:
If we use sections in the json schema, we're going to have to spread the formData at each nesting level, which I think means we can't entirely get away from FE data transformation

Reply:
Curious aren't you doing this in reverse when you load the formData that is returned from the database? I haven't tried this but by default `omitExtraData` is set to true in `FormBase` so it might even clean it for us automatically.

```
const newFormData = {
section1: {
     ...formData
},
section2: {
     ...formData
},
section3: {
     ...formData
},
}

 The only thing with that is we would need to be careful not to have fields named the same at the root of each nested section.

Reply:
I am doing it in reverse, but in the ninja schema instead of in the front end. I wanted to do everything in the ninja schema but can't here because of the nesting.